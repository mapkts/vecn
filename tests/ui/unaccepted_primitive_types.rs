#[vecn::vector]
struct TupleStruct1(usize, usize);

#[vecn::vector]
struct TupleStruct2(isize, isize);

#[vecn::vector]
struct TupleStruct3(bool, bool);

#[vecn::vector]
struct NamedStruct1 {
    x: usize,
    y: usize,
}

#[vecn::vector]
struct NamedStruct2 {
    x: isize,
    y: isize,
}

#[vecn::vector]
struct NamedStruct3 {
    x: bool,
    y: bool,
}

fn main() {}
