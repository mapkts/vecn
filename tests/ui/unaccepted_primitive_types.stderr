error: expected numeric primitive type, but found others
 --> $DIR/unaccepted_primitive_types.rs:2:20
  |
2 | struct TupleStruct1(usize, usize);
  |                    ^^^^^^^^^^^^^^

error: expected numeric primitive type, but found others
 --> $DIR/unaccepted_primitive_types.rs:5:20
  |
5 | struct TupleStruct2(isize, isize);
  |                    ^^^^^^^^^^^^^^

error: expected numeric primitive type, but found others
 --> $DIR/unaccepted_primitive_types.rs:8:20
  |
8 | struct TupleStruct3(bool, bool);
  |                    ^^^^^^^^^^^^

error: expected numeric primitive type, but found others
  --> $DIR/unaccepted_primitive_types.rs:11:21
   |
11 |   struct NamedStruct1 {
   |  _____________________^
12 | |     x: usize,
13 | |     y: usize,
14 | | }
   | |_^

error: expected numeric primitive type, but found others
  --> $DIR/unaccepted_primitive_types.rs:17:21
   |
17 |   struct NamedStruct2 {
   |  _____________________^
18 | |     x: isize,
19 | |     y: isize,
20 | | }
   | |_^

error: expected numeric primitive type, but found others
  --> $DIR/unaccepted_primitive_types.rs:23:21
   |
23 |   struct NamedStruct3 {
   |  _____________________^
24 | |     x: bool,
25 | |     y: bool,
26 | | }
   | |_^
